packages/create-next-app/templates/index.ts

```diff
@@ -301,6 +301,13 @@ export const installTemplate = async ({
   const devDeps = Object.keys(packageJson.devDependencies).length;
   if (!devDeps) delete packageJson.devDependencies;
 
+  /* For pnpm v10+ x-ref: https://github.com/vercel/next.js/issues/83158 */
+  if (packageManager === "pnpm") {
+    packageJson.pnpm = {
+      onlyBuiltDependencies: ["sharp"],
```

Comment:
```suggestion
      ignoredBuiltDependencies: ["sharp"],
```

I misread the setting and thought it was the same as my PR I was about to create.

Instead, we should use `ignoredBuiltDependencies` to simply ignore the script.

We definitely don't want to auto approve it 😓

Reply:
@styfle seems like you're aiming for a more security-forward stance here.

I think this inconsistent behavior could be likely to cause unknown pnpm-only breakages in future (or on [certain platforms, like with `sharp`](https://github.com/vercel/next.js/pull/83168#pullrequestreview-3356842471)), when a `postinstall` script is indeed needed.

Since npm and Yarn do not warn on `postinstall` scripts, I think the easiest fix is to make pnpm less noisy, automatically approving the build via the package `postinstall` scripts like other package managers (the original way that this PR was created). This will make the experience identical on all package managers.

However, if the Next.js team wants to take a more strict, security-forward stance, my suggestion would be again to make the experience the same for all package managers and add it as an interactive question + flag to ask if users would like to build the packages:

- https://github.com/vercel/next.js/pull/83168#discussion_r2358839702

Since @eps1lon mentioned that he was going to bring it up with the Next.js security team, maybe you could surface this again with this team?

Reply:
@styfle also note that Bun already includes a list of default trusted dependencies to be built, which includes `sharp`:

> Bun includes a default allowlist of popular packages containing `postinstall` scripts that are known to be safe. You can see this list here.

- https://github.com/oven-sh/bun/blob/b278c8575363665fda9f2c449d9691872051a1d7/src/install/default-trusted-dependencies.txt#L308
- Docs: https://bun.com/guides/install/trusted

Reply:
We're currently not in a position to curate a list of trusted dependencies and their versions. `ignoreBuiltDependencies` is sufficient here and avoids having to audit these packages.